### PR TITLE
Symfony workflow

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: src/Bundle/Controller/Workflow.php
Comment (loevgaard) [inline src/Bundle/Controller/Workflow.php:31]: What are these inheritdoc comments used for? I have always wondered, never found an answer

Comment (vvasiloi) [reply src/Bundle/Controller/Workflow.php:31]: It means the documentation is inherited from parent.

Comment (loevgaard) [reply src/Bundle/Controller/Workflow.php:31]: Yes. My comment was too vague. I meant: What tooling uses these 'inheritdocs'? I never add these and I haven't heard anyone wanting to add them - on the contrary in fact.

Comment (loevgaard) [reply src/Bundle/Controller/Workflow.php:31]: Just saw this commit right now: https://github.com/Sylius/SyliusResourceBundle/commit/369bfadc73c6b49011ab058397e4791418a6ab1c. Looks like I am not the only one wondering

Comment (vvasiloi) [reply src/Bundle/Controller/Workflow.php:31]: It's used by IDEs and code assistance tools like PHPSTORM and https://serenata.gitlab.io/ and by documentation generators like https://www.phpdoc.org/.
You can see it in [PSR-19 Draft](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md#42-using-the-inheritdoc-inline-tag-to-augment-a-description), which accompanies [PSR-5 Draft](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md) in an attempt to make a standard for PHP doc blocks. 
Similar notation is used in Java, JS, XML and probably more.

Comment (vvasiloi) [reply src/Bundle/Controller/Workflow.php:31]: image
Here's an example where it can be useful, but adding it everywhere makes no sense.

File: src/Bundle/DependencyInjection/Compiler/RegisterStateMachinePass.php
Comment (loic425) [inline src/Bundle/DependencyInjection/Compiler/RegisterStateMachinePass.php:29]: @pamil I have to add a phpunit test for this compiler pass.

**Comment by loic425** [comment]
@pamil @lchrusciel @Zales0123 I'm waiting for your review. 🤞

**Comment by loic425** [comment]
@pamil Just to note here what I said to you on private message.

Do you want me to execute [these tests](https://github.com/Sylius/SyliusResourceBundle/blob/58fbc66265f2b4c210d2a4b9b7c99a1740e508fe/src/Bundle/test/src/Tests/Controller/PullRequestApiTest.php) with a winzou state machine configuration.

File: .github/workflows/build.yml
Comment (loic425) [inline .github/workflows/build.yml:105]: Maybe using an env var directly on config should be lot better.

**Comment by loic425** [comment]
@pamil Winzou state machine is now tested with same phpunit tests.

File: docs/state_machine.md
Comment (loic425) [reply docs/state_machine.md:42]: You're right, I will make that changes too...

File: .github/workflows/build.yml
Comment (loic425) [reply .github/workflows/build.yml]: @pamil yes, done. And I inverted the sequence (winzou by default)

File: docs/state_machine.md
Comment (loic425) [reply docs/state_machine.md]: Done!

**Comment by loic425** [comment]
@pamil I've just finished refactoring this PR with new test app architecture.

**Comment by loic425** [comment]
@pamil @lchrusciel @AdamKasp
Now 1.7 has been released, can we finish this PR?

**Comment by loic425** [comment]
ping @lchrusciel @pamil

File: docs/state_machine.md
Comment (loic425) [reply docs/state_machine.md]: This is done...

**Comment by Zales0123** [comment]
Hi, @loic425 😄 As you're doing great with PRs today :dancer: maybe it would be worth returning to this one? I will dig into it and do some tests, but it seems promising and finally mergeable in the very nearest future 🖖

**Comment by loic425** [comment]
> Hi, @loic425 😄 As you're doing great with PRs today 💃 maybe it would be worth returning to this one? I will dig into it and do some tests, but it seems promising and finally mergeable in the very nearest future 🖖

done 👍

**Review by Zales0123 — COMMENTED** [review]
Some few final comments and it will be ready to merge finally :) 🖖 Especially [this one](https://github.com/Sylius/SyliusResourceBundle/pull/203#discussion_r781373010) is important

File: src/Bundle/Tests/DependencyInjection/Compiler/RegisterStateMachinePassTest.php
Comment (loic425) [reply src/Bundle/Tests/DependencyInjection/Compiler/RegisterStateMachinePassTest.php:29]: done

File: src/Bundle/DependencyInjection/Compiler/RegisterStateMachinePass.php
Comment (loic425) [reply src/Bundle/DependencyInjection/Compiler/RegisterStateMachinePass.php:87]: done, you're right! thx :)

File: src/Bundle/Controller/Workflow.php
Comment (loic425) [reply src/Bundle/Controller/Workflow.php]: done

and on apply method too...

File: src/Bundle/spec/Controller/WorkflowSpec.php
Comment (loic425) [reply src/Bundle/spec/Controller/WorkflowSpec.php:32]: IMHO I think it's cool to keep it cause you have a quick access to the class you describes :)

File: src/Bundle/test/config/packages/test/framework.yaml
Comment (loic425) [reply src/Bundle/test/config/packages/test/framework.yaml:4]: It's on the Github actions.
https://github.com/Sylius/SyliusResourceBundle/blob/516b67989b9a70b0f358c0ec30d42e68598b47b0/.github/workflows/build.yml#L95

File: src/Bundle/spec/Controller/WorkflowSpec.php
Comment (loic425) [reply src/Bundle/spec/Controller/WorkflowSpec.php]: done

File: src/Bundle/test/src/Entity/PullRequest.php
Comment (loic425) [reply src/Bundle/test/src/Entity/PullRequest.php]: done and I update the id property too.

File: src/Bundle/test/src/Form/Type/PullRequestType.php
Comment (loic425) [reply src/Bundle/test/src/Form/Type/PullRequestType.php]: done

File: src/Bundle/test/src/Form/Type/PullRequestType.php
Comment (loic425) [reply src/Bundle/test/src/Form/Type/PullRequestType.php]: done

File: src/Bundle/test/src/Tests/Controller/PullRequestApiTest.php
Comment (loic425) [reply src/Bundle/test/src/Tests/Controller/PullRequestApiTest.php]: done

File: src/Bundle/test/config/packages/test/framework.yaml
Comment (Zales0123) [reply src/Bundle/test/config/packages/test/framework.yaml:4]: I mean - in symfony/workflow you can have a workflow with `state_machine` type and `workflow` type. The other one allows multiple states, as far as I understand and, as far as I checked, may not work with the current implementation... or have I misunderstood this GitHub actions step and this is what it does? 😄

Comment (loic425) [reply src/Bundle/test/config/packages/test/framework.yaml:4]: No you're right, I misunderstood the config you pointed.
I was thinking about the sylius_resource config.

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Finally :tada: Thank you, @loic425, astonishing work! :rocket:

**Pull request merged**
